runtime/lua/vim/lsp/util.lua

```diff
@@ -924,21 +924,25 @@ function M.locations_to_items(locations)
   for _, d in ipairs(locations) do
     -- locations may be Location or LocationLink
     local uri = d.uri or d.targetUri
-    local fname = assert(vim.uri_to_fname(uri))
     local range = d.range or d.targetSelectionRange
-    table.insert(grouped[fname], {start = range.start})
+    table.insert(grouped[uri], {start = range.start})
   end
 
 
   local keys = vim.tbl_keys(grouped)
   table.sort(keys)
   -- TODO(ashkan) I wish we could do this lazily.
-  for _, fname in ipairs(keys) do
-    local rows = grouped[fname]
+  for _, uri in ipairs(keys) do
+    local rows = grouped[uri]
 
     table.sort(rows, position_sort)
     local i = 0
-    for line in io.lines(fname) do
+    local bufnr = vim.uri_to_bufnr(uri)
+    if not api.nvim_buf_is_loaded(bufnr) then
+      vim.fn.bufload(bufnr)
```

Comment:
Should the buffers that are loaded be unloaded afterwards again?

Reply:
Hmm... I think leave them loaded seems fine. This would prevent us from having to read through them all the time when you get the information, like we were doing before.

Reply:
Hmm... although if they changed in the background or something, that could be kind of annoying. I don't know if that's a concern that we should have or not.

Reply:
> although if they changed in the background or something, that could be kind of annoying. I don't know if that's a concern that we should have or not.

Also not sure. Maybe keep it loaded for now and if some issues pop up we change it to unload buffers that were implicitly loaded?

Reply:
Maybe just add an "else vim.cmd [[checktime]]"?

Reply:
> Maybe just add an "else vim.cmd [[checktime]]"?

done

Reply:
Actually, what I think is better is just calling bufload on it always. Since bufload will read the file if it's already loaded. So we can just skip the check for if it's loaded and not do the checktime.

Reply:
> Since bufload will read the file if it's already loaded.

Ah nice, wasn't aware of that. Pushed the change.